Sources/Kitura/CodableRouter.swift

```diff
@@ -600,21 +603,37 @@ extension Router {
         }
     }
 
-    internal func parameterIsPresent(in route: String) -> Bool {
-        if route.contains(":") {
-            let paramaterString = route.split(separator: ":", maxSplits: 1, omittingEmptySubsequences: false)
-            let parameter = paramaterString.count > 0 ? paramaterString[1] : ""
-            Log.error("Erroneous path '\(route)', parameter ':\(parameter)' is not allowed. Codable routes do not allow parameters.")
+    /// Precondition: Path is known to contain a : character
+    func pathHasSingleParamIdAsSuffix(_ path: String) -> Bool {
```

Comment:
As this function is only used from `pathSyntaxIsValid()` can we make it a [nested function](https://docs.swift.org/swift-book/LanguageGuide/Functions.html#ID178)?

Reply:
I agree ..however, I would feel that it may readability.  How about marking it private and restricting its access ?

Reply:
I don't see how it would affect readability. This is what nested functions are for - small helpers used by another function.